File: Try_GameEngine5/src/com/example/try_gameengine/remotecontroller/RightKeyCommandPressUp.java
package com.example.try_gameengine.remotecontroller;

import android.view.MotionEvent;

import com.example.try_gameengine.remotecontroller.RemoteController.CommandType;

public class RightKeyCommandPressUp implements Command{
	RightKey rightKey;
	private int pointerId = -1;
	
	public RightKeyCommandPressUp(RightKey rightKey){
		this.rightKey = rightKey;
	}
	
	@Override
	public CommandType execute() {
		return CommandType.RightKeyUpCommand;
	}

	@Override
	public boolean checkExecute(float x, float y, MotionEvent event) {
		return rightKey.pressUp(x, y, event);
	}

	@Override
	public void setMotionEventPointerId(int pointerId) {
		this.pointerId = pointerId;
	}

	@Override
	public int getMotionEventPointerId() {
		return pointerId;
	}

}

File: Try_GameEngine5/src/com/example/try_gameengine/remotecontroller/RightKeyCommandPressUp.kt
package com.example.try_gameengine.remotecontroller

import android.view.MotionEvent

class RightKeyCommandPressUp(rightKey: RightKey) : Command {
    var rightKey: RightKey
    private var pointerId = -1

    init {
        this.rightKey = rightKey
    }

    override fun execute(): RemoteController.CommandType {
        return RemoteController.CommandType.RightKeyUpCommand
    }

    override fun checkExecute(x: Float, y: Float, event: MotionEvent?): Boolean {
        return rightKey.pressUp(x, y, event)
    }

    override fun setMotionEventPointerId(pointerId: Int) {
        this.pointerId = pointerId
    }

    override fun getMotionEventPointerId(): Int {
        return pointerId
    }
}
